### PR TITLE
School library class associations

### PR DESCRIPTION
**Pull request by simplegoose**
In this PR, the following was done:

- A classroom file was created classroom file, and inside the classroom file, a Classroom class was created.
- Class student was modified to allow initialize method to accept a classroom object, permission and name.
- A new class inside book.rb was created, and its variables were initialized.
- A new class rental inside rental.rb was created, and its variables were initialized.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (1GORDON) [inline book.rb]: Great job @simplegoose 🤝

- [ ] Kindly ensure a method for adding rentals should be present. `The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.`. You may follow this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more clarity.

File: person.rb [hunk 2]
```diff
@@ -15,10 +15,11 @@ def initialize(
     @id = Random.rand(1...1000)
     @name = name
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (1GORDON) [inline person.rb:22]: Great job @simplegoose 🤝

- [x] Kindly ensure a method for adding rentals should be present. `The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association`. You may follow this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more clarity.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @simplegoose  ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] Could you fix the broken issues link to make your readme file more professional?

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (simplegoose) [reply book.rb]: Hello @1GORDON ,
Thank you for the review, the functionality has been achieved in the Rental class. I have used a method similar to the one given in the documentation. Can you please check? Or if it is not the correct method please let me know, Thank you.

Example used:
```
class Animal
  attr_reader :owner, :visits

  def initialize(type, number_of_legs, name = "Unknown")
    @id = Random.rand(1..1000)
    @name = name
    @number_of_legs = number_of_legs
    @type = type
    @liked_food = NoFood.new()
    @visits = []
  end

  ...
end
```
```
class Visit
  attr_reader :animal, :vet
  attr_accessor :date

  def initialize(date, animal, vet)
    @date = date

    @animal = animal
    animal.visits << self

    @vet = vet
    vet.visits << self
  end
end
```

Comment (danmainah) [reply book.rb]: - [x] @simplegoose  great work implementing the changes requested just to clarify the rental class implement the belongs_to relationship and the book class implements the has_many association, so you require both to achieve the association

**Review by danmainah — APPROVED** [review]
Hi @simplegoose , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Linter setup correctly and all checks are passing
✔️ All associations added successfully

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**